### PR TITLE
Staging grpc v2.5

### PR DESCRIPTION
**Pull request by lgarner**
Initial GRPC version with events.

### CODE REVIEW

**Review by brucearmstrong — APPROVED** [review]
Looks pretty straightforward.

File: Runtime/Scripts/MobiledgeXIntegrationHelper.cs [hunk 1]
```diff
@@ -227,16 +228,21 @@ public async Task<bool> FindCloudlet(string dmeHost = null, uint dmePort = 0)
...
             latestFindCloudletReply = reply;
-            latestAppPortList = reply.ports;
-            return reply.status == FindCloudletReply.FindStatus.FIND_FOUND;
+            int porti = 0;
+            latestAppPortList = new AppPort[reply.Ports.Count];
+            foreach (var aport in reply.Ports)
+            {
+               latestAppPortList[porti++] = aport;
+            }
+            return reply.Status == FindCloudletReply.Types.FindStatus.FindFound;
```
Comment (lgarner) [inline Runtime/Scripts/MobiledgeXIntegrationHelper.cs:245]: BTW, we really ought to return the actual enum back in REST and GRPC, it's useful, versus the boolean.

File: Runtime/Scripts/MobiledgeXIntegration.cs [hunk 2]
```diff
@@ -77,7 +77,7 @@ public partial class MobiledgeXIntegration
         bool latestVerifyLocationStatus = false; // Whether the most recent verifyLocation call was successful
         FindCloudletMode mode = FindCloudletMode.PROXIMITY; // FindCloudlet mode
         AppPort latestAppPort = null;
-        AppPort[] latestAppPortList = null;
+        AppPort [] latestAppPortList = null;
```
Comment (franklin-huang-mobiledgex) [inline Runtime/Scripts/MobiledgeXIntegration.cs:80]: nit extra space

File: Runtime/Scripts/MobiledgeXIntegrationHelper.cs (hunk 1 above)
Comment (franklin-huang-mobiledgex) [reply Runtime/Scripts/MobiledgeXIntegrationHelper.cs:245]: Why? We do return in c#, but i'm guessing unity devs only care about if the findcloudlet was successful or not. Plus, returning FIND_NOTFOUND or FIND_UNKNOWN isn't very helpful in this case

**Review by franklin-huang-mobiledgex — APPROVED** [review]

File: Runtime/Scripts/MobiledgeXIntegrationHelper.cs (hunk 1 above)
Comment (lgarner) [reply Runtime/Scripts/MobiledgeXIntegrationHelper.cs:245]: It's not a guarantee there will never have new states (right now, it is mostly just the 2 states, Unknown shouldn't happen outside a Unity AOT reflection error (link.xml, etc), for example), and a switch statement isn't much more difficult to handle.

**Pull request closed without merging**